packages/elements/src/components/Spreadsheet/index.tsx

```diff
@@ -52,6 +53,32 @@ export const AddRowButton = ({ addNewRow }) => {
   )
 }
 
+export const genarateErrorElements = (data: Cell[][] = []) => {
+  const errors: JSX.Element[] = []
```

Comment:
Looks good. I think you may want to extract few complex parts by either adding comments or extract to a function to reduce code complexity